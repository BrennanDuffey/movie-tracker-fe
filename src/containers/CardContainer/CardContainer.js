import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SmallCard from '../SmallCard/SmallCard';
import butterflixMascot from '../../images/character.svg';

export class CardContainer extends Component {
  constructor() {
    super()
    this.state = {
    }
  }
  render() {
    let smallCards;
    !this.props.movies.length
      ? smallCards = null 
      : smallCards = this.props.movies.map(movie=>
          (<SmallCard movie={movie} key={movie.id} />)
        )

    return (
      <main className="CardContainer">
        <header>
          <div className="title">
            <img src={butterflixMascot} alt="butterflix-fairy-mascot"/>
            <h1>ButterFlix</h1>
          </div>
        </header>
        <div className="display-divider"></div>
        <section className="cards-display">
          {smallCards}
        </section>
      </main>
    )
  }
}

CardContainer.propTypes={
  history: PropTypes.object,
  location: PropTypes.object,
  match: PropTypes.object,
  movies: PropTypes.array
}

export const mapStateToProps = (state) => ({
  movies: state.movies
})

export default connect(mapStateToProps)(CardContainer);