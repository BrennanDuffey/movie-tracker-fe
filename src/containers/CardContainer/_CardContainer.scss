.CardContainer {
  position: relative;
  box-sizing: border-box;
  width: 90vw;
  height: min(100vh);
  display: grid;
  grid-template-rows: 10% 90%;
  background-color: #3a3a3a;
  background: url('https://images.unsplash.com/photo-1489599849927-2ee91cede3ba?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80');
  background-size: cover;
  backdrop-filter: 80%;
}

.display-divider {
  content: '';
  top: 9.5%;
  left: 1%;
  position: absolute;
  width: 95%;
  height: 5px;
  z-index: 4;
  background-color: #500000;
  border-radius: 40%;
  border-bottom: 3px solid rgba(0, 0, 0, 0.2);
  box-shadow: 0 4px 5px black;
}

header {
  background-color: black;
  display: flex;
  justify-content: flex-end;
  // justify-content: center;
  align-items: center;
  padding: 20px 5%;

    // .search-icon {
    //   position: relative;
    //   display: flex;
    //   justify-content: center;
    //   align-items: center;
    //   z-index: 3;
    //   content: '';
    //   height: 50px;
    //   width: 50px;
    //   background-color: #500000;
    //   box-shadow: 2px 2px 10px rgba(157, 157, 157, 0.4);
    //   border-radius: 50%;

    //     .fa-search {
    //       color: silver;
    //     }
    // }


    .title {
      position: absolute;
      display: flex;
      justify-content: center;
      align-items: center;
      left: 15px;

        h1 {
          color: #970000;
          font-size: 34px;
        }
    }
    img {
      width: 30px;
      height: 30px;
      margin-right: 5px;
    }
}

.cards-display {
  display: flex;
  flex-wrap: wrap;
  padding: 0 max(5%);
  // margin: 50px max(50px);
  overflow: scroll;
}

.SmallCard {
  display: grid;
  grid-template-rows: 60% 40%;
  align-items: center;
  position: relative;
  background: rgba(0, 0, 0, 0.51);
  box-shadow: 4px 4px 10px rgba(0, 0, 0, 0.585);
  width: 250px;
  height: 356px;

    .image--container {
      content: '';
      position: absolute;
      height: 100%;
      width: 100%;
      background-color: silver;
      z-index: 1;
      
        img {
          object-fit: cover;
          width:100%;
          height: 100%;
        }
    }

    // & > .content:hover {
    //       display: none;
    //     }

    .content {
      color: white;
      background: linear-gradient(rgba(0, 0, 0, 0.8), rgba(100,0,0,6));
      border-radius: 20% 20% 0 0;
      padding: 15px;
      z-index: 3;
      position: absolute;
      bottom: 0px;
      height: 40%;

        h3 {
          display: flex;
          justify-content: center;
          align-items: center;
          height: 50px;
          margin: 0;
          text-align: center;
          text-shadow: 2px 2px 3px black;
          overflow: scroll;
        }

        p {
          margin-bottom: 0;
          text-shadow: 1px 1px 3px black;
        }

        .sub-content {
          position: relative;
          display: flex;
          align-items: flex-end;
          flex-direction: column;

            .sub-border {
              content: '';
              width: 100%;
              height: 3px;
              box-shadow: 1px 1px 3px black;
              background-color: #500000;
              border-radius: 40%;
            }

            .showmore {
              text-align: right;
            }
        }
    }

    .rating {
      position: absolute;
      bottom: 10px;
      left: 10px;
      display: flex;
      justify-content: center;
      align-items: center;
      font-size: 20px;
      background: #970000;
      box-shadow: 0px 4px 4px #000000;
      width: 35px;
      height: 35px;
      border-radius: 50%;
      color: white;
    }

    .faveBtn {
      position: absolute;
      z-index: 5;
      color: silver;
      font-size: 25px;
      bottom: 10px;
      left: 50px;
      background: none;
      border: none;

      &:hover {
        color: rgb(232, 181, 27);
      }
    }
}