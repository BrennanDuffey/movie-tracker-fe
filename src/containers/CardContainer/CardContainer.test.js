import {CardContainer} from './CardContainer';
import React from 'react';
import {shallow} from 'enzyme';

let mockMovies = [
  { title: "movie", poster: "http", backdrop: "http", summary: "words",
    genres: ["action"], release: 2000, rating: 9.9, id: 69, isFavorite: true
  },
  { title: "nextMovie", poster: "https", backdrop: "https", summary: "MoreWords",
    genres: ["drama"], release: 2001, rating: 9.2, id: 69, isFavorite: true
  }
];

describe('CardContainer', ()=>{
  let wrapper;

  it('should render a container w/ no cards if passed no movies', ()=>{
    wrapper= shallow(<CardContainer movies={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a container w/ cards if passed movies', ()=>{
    wrapper= shallow(<CardContainer movies={mockMovies}/>);
    expect(wrapper).toMatchSnapshot();
  })
});