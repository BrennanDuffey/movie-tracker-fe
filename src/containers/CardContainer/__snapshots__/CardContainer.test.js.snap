// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should render a container w/ cards if passed movies 1`] = `
<main
  className="CardContainer"
>
  <header>
    <div
      className="title"
    >
      <img
        alt="butterflix-fairy-mascot"
        src="character.svg"
      />
      <h1>
        ButterFlix
      </h1>
    </div>
  </header>
  <div
    className="display-divider"
  />
  <section
    className="cards-display"
  >
    <ConnectFunction
      key="69"
      movie={
        Object {
          "backdrop": "http",
          "genres": Array [
            "action",
          ],
          "id": 69,
          "isFavorite": true,
          "poster": "http",
          "rating": 9.9,
          "release": 2000,
          "summary": "words",
          "title": "movie",
        }
      }
    />
    <ConnectFunction
      key="69"
      movie={
        Object {
          "backdrop": "https",
          "genres": Array [
            "drama",
          ],
          "id": 69,
          "isFavorite": true,
          "poster": "https",
          "rating": 9.2,
          "release": 2001,
          "summary": "MoreWords",
          "title": "nextMovie",
        }
      }
    />
  </section>
</main>
`;

exports[`CardContainer should render a container w/ no cards if passed no movies 1`] = `
<main
  className="CardContainer"
>
  <header>
    <div
      className="title"
    >
      <img
        alt="butterflix-fairy-mascot"
        src="character.svg"
      />
      <h1>
        ButterFlix
      </h1>
    </div>
  </header>
  <div
    className="display-divider"
  />
  <section
    className="cards-display"
  />
</main>
`;
